editor/components/block-list/block.js

```diff
@@ -504,6 +480,7 @@ const mapStateToProps = ( state, { uid } ) => ( {
 	isFirstMultiSelected: isFirstMultiSelectedBlock( state, uid ),
 	isHovered: isBlockHovered( state, uid ) && ! isMultiSelecting( state ),
 	focus: getBlockFocus( state, uid ),
+	shouldFocusNext: shouldBlockFocusNext( state, uid ),
```

Comment:
So if I understand properly this new prop is useful because there's a new `dispatch` that will be performed later to move the focus? Why can't we batch these two dispatches together?

 - Maybe a unique actions
 - Using a store enhancer or a middleware (I believe these things already exist)

Reply:
Yes, we let the editor that needs focus call `onFocus` by itself later. There's no newly added dispatch here, we are just moving focus too early. I'm not sure how an enhancer or middleware would help.